### DISCUSSION TITLE
Garage Management

### DISCUSSION THREAD

**Discussion by RippedTrojan** [RoboSats/robosats#386] [Q&A] [answered]
Greetings! I've spent the last several days learning the in's and out's of the platform and have been having lots of fun. I've recently begun exploring the ability to have multiple orders active at the same time and I have a couple of questions. 

Building the Garage is very straight forward and couldn't be much simpler. Where I face issues is _recovering_ my garage. I see that I can delete my entire garage, I can download my entire garage...but I haven't figured out if there is a simple way to _restore_ my entire garage. I am running everything on a laptop and will often disconnect from the internet or have to close things down for mobility purposes. When I get back up and running, my work flow looks something like this:

1. Must restore garage for whatever reason 
2. Use "Recover" button to restore first bot with it's token
3. Click add bot
4. Once the new bot loads, log it out
5. Use the now visible "recover" button to recover second token
6. Repeat steps 3 through 5 until entire garage is recovered.

If I skip steps 2 and 3, I am forced to logout of a bot with an active order before I have access to the "Recover" button. Is there any easier way to do this?

**Comment by Reckless-Satoshi** [comment, marked as answer]
Hey @RippedTrojan , glad to hear you are having fun :D

>     1. Must restore garage for whatever reason
> 
>     2. Use "Recover" button to restore first bot with it's token
> 
>     3. Click add bot
> 
>     4. Once the new bot loads, log it out
> 
>     5. Use the now visible "recover" button to recover second token
> 
>     6. Repeat steps 3 through 5 until entire garage is recovered.

You have figured out the best possible way to reconstruct a garage at the moment. The download button is more of an "emergency backup" rather than a convenience tool. There is no "load from file" option yet. Eventually, this feature will come. In fact, in the garage file we could save presets for the maker form, app settings, and even pre-formatted messages to auto-send when the chat opens! None of these will come until we work on more features of RoboSats Pro (live concept at unsafe.robosats.com/pro).

Another cool thing we could do with the Garage is to add the "extended token". A single token from which every robot is derived given an index.
